File: hw2/p2.R
#########
# Setup #
#########

library(MASS)
library(klaR)

data <- data.frame(
    c(0.6585, 2.2460, -2.7665, -1.2565, -0.7973, 1.1170),
    c(0.2444, 0.5281, -3.8303, 3.4912, 1.2288, 2.2637),
    c(0, 0, 0, 1, 1, 1)
)

test <- data.frame(c(0), c(1))

names(data) <- c("F1", "F2", "CLASS")
names(test) <- c("F1", "F2")

train.size <- nrow(data)
train <- data[1:train.size, 1:2]
train.cl <- factor(data[1:train.size, 3])
 
#######
# QDA #
#######

model <- qda(x = train, grouping = train.cl)
predict <- predict(model, test)
print(predict)
print(paste("QDA: ", predict$class))

# Posterior Class Probabilities
# 0: 0.0067895
# 1: 0.9932105

###############
# Naive Bayes #
###############

model <- NaiveBayes(x = train, grouping = train.cl, usekernel=FALSE)
predict <- predict(model, test)
print(predict) 
print(paste("Naive Bayes: ", predict$class))

# Posterior Class Probabilities
# 0: 0.2493135
# 1: 0.7506865

File: hw2/p2.m
function p2

%% Data
% Input features to train on.
train = [
    % Class 0
    0.6585, 0.2444;
    2.2460, 0.5281;
    -2.7665, -3.8303;
    % Class 1
    -1.2565, 3.4912;
    -0.7973, 1.2288;
    1.1170, 2.2637
];
% Labels for the input features.
labels = [0; 0; 0; 1; 1; 1];
% We want to know the class and
% posterior class probabilities for this point.
test = [0, 1];

%% Classification
% Create our model, and use it to evaluate the test point.
model = fitcnb(train, labels, 'Distribution', 'normal')
% label = 1
% Posterior = 0.2493, 0.7507
[label, Posterior] = predict(model, test)

end